chore: upgrade storybook

koji opened the pull request:
<!--
Thanks for taking the time to open a pull request! Please make sure you've read the "Opening Pull Requests" section of our Contributing Guide:

https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

To ensure your code is reviewed quickly and thoroughly, please fill out the sections below to the best of your ability!
-->

# Overview
This PR upgrade storybook and 3 addons.
<!--
Use this section to describe your pull-request at a high level. If the PR addresses any open issues, please tag the issues here.
-->

# Changelog
- update package.json and yarn.lock file
<!--
List out the changes to the code in this PR. Please try your best to categorize your changes and describe what has changed and why.

Example changelog:
- Fixed app crash when trying to calibrate an illegal pipette
- Added state to API to track pipette usage
- Updated API docs to mention only two pipettes are supported

IMPORTANT: MAKE SURE ANY BREAKING CHANGES ARE PROPERLY COMMUNICATED
-->

# Review requests
- [ ] `make -C components dev` works properly
<!--
Describe any requests for your reviewers here.
-->

# Risk assessment
low
<!--
Carefully go over your pull request and look at the other parts of the codebase it may affect. Look for the possibility, even if you think it's small, that your change may affect some other part of the system - for instance, changing return tip behavior in protocol may also change the behavior of labware calibration.

Identify the other parts of the system your codebase may affect, so that in addition to your own review and testing, other people who may not have the system internalized as much as you can focus their attention and testing there.
-->

koji committed:
chore: upgrade storybook

This PR upgrade storybook and 3 addons

koji committed:
chore: upgrade storybook

This PR upgrade storybook and 3 addons

koji committed:
Merge branch 'edge' into chore_app-update-storybook

vabruzzo committed:
refactor(app): remove need for hardcoded heights in module grids (#11146)

jbleon95 committed:
feat(engine): heater-shaker movement restrictions (#10861)

Adds a new method to check if pipette movement is allowed when a heater-shaker is on the deck, raising an error if the movement is not allowed.

SyntaxColoring committed:
test(update-server): Make minor fixes to tests (#11150)

* Delete accidentally duplicated function.

* Work around TestClient silliness.

> PytestCollectionWarning: cannot collect test class 'TestClient' because it has a __init__ constructor

SyntaxColoring committed:
test(update-server): Start type-checking tests (#11151)

Commit:
refactor(app): update H-S menu deactivate copy (#11102)

This PR updates the "deactivate" copy to "deactivate heater" for better clarity.

closes #11059

TamarZanzouri committed:
Fixed failing CI tests after merging feature branch into edge (#11157)

brenthagen reviewed (commented):
there are a few additional `yarn.lock` changes on install, probably from merging in edge. otherwise tested this out, and also smoke tested launching the app, labware library, and protocol designer, and don't see any issues.

Commit:
refactor(app): update H-S TestShake slideout and overflow menu (#11132)

* refactor(app): update H-S TestShake slideout and overflow menu

This PR updates the TestShake slideout and updates the overflow menu selection. It also moves the labware latch status below the title. Additionally, it addresses the tooltip bug on the slideout. closes #11074 closes #11136 closes #11045

koji committed:
refactor(app): this will fix 11010 (#11155)

This will remove css={{}} for overflow-wrap
fix #11010

koji committed:
refactor(app): App use textTransform from typography (#10911)

use textTransform property from TYPOGRAPHY

koji committed:
fix(app): Fix text spacing issue on LPC screen (#11159)

This will fix the text spacing issue on LPC screen.
fix #110671

jerader commented on package.json:
```diff
@@ -45,10 +45,10 @@
-    "@storybook/addon-essentials": "^6.2.5",
-    "@storybook/addon-links": "^6.2.5",
-    "@storybook/react": "^6.2.5",
+    "@storybook/addon-actions": "^6.5.9",
+    "@storybook/addon-essentials": "^6.5.9",
+    "@storybook/addon-links": "^6.5.9",
+    "@storybook/react": "^6.5.9",
```
I have a quick question, I thought that the `^` means that it will automatically update to the future patch versions. So `^6.2.5` actually updates to `^6.5.9` already?

koji replied:
`^`
>Allows changes that do not modify the left-most non-zero digit in the [major, minor, patch] tuple.

^1.2.3 := 1.2.3 <= version < 2.0.0
^0.1.2 := 0.1.2 <= version < 0.2.0
^0.0.1 := 0.0.1 <= version < 0.0.2

^6.2.5<= version 7.0.0

the version depends on the timing we update. I have another opentrons repo on mac and the version of storybook is `6.2.9`.


For this update, I used `npx sb upgrade`

jerader committed:
feat(app): show text if no modules plugged in but required during setup (#11140)

closes #10947

MichaelMulieri committed:
fix(app): update hover state of historical run chevron (#11153)

brenthagen committed:
refactor(app): remove js resize listeners for card grids (#11169)

this refactors a previous card layout bug fix that added js listeners, directly substituting css
media queries for the js

re #10886

jerader committed:
refactor(app): h-S modal update hover states and checkbox to match designs (#11168)

closes #11162

smb2268 committed:
feat(app, components): align Liquid Setup UI with Figma (#11044)

* feat(app, components): align Liquid Setup UI with Figma

brenthagen committed:
fix(app): add firstNonSetupIndex to historical run commands query cursor (#11158)

this omits the non setup commands from the commands query for historical runs, which was the cause
of some run commands at the end of the run log not being fetched

closes #11135

jerader committed:
refactor(app): clean up H-S and ModduleSetup Banner components (#11192)

vabruzzo committed:
fix(app): fixes extra module bug in module controls (#11195)

koji committed:
refactor(app): refactor Robotsettings networktab for isrobotbusy (#11176)

This PR will remove useIsRobotBusy from network select component.
fix #11128

koji committed:
refactor(app): this will fix #11186 (#11197)

This PR will clean up robot_settings.json file
fix #11186

TamarZanzouri committed:
fix(PAPIv2): allow presses=0 to hover over tip when calling pick_up_tip (#11200)

ecormany committed:
update broken and redirecting support links (#11190)

jerader committed:
refactor(app): h-S refactor module command logic in wizard to work in terminal state (#11185)

closes #11179

TamarZanzouri committed:
fix(papi-v2): allow pick_up_tip to pick up from exact location  (#11038)

koji committed:
refactor(app): refactor Robotsettings Advancedtab for isRobotBusy (#11188)

This pr changes the component that calls useIsBusyRobot from each RobotSettingsAdvance child
component from RobotSettingsAdvance component and useIsBusyRobot fetches the status every 30sec.
fix #11128

DerekMaggio committed:
feat(api): Heater Shaker Firmware Level Emulator (#11121) (#11207)

* feat: Heater-Shaker Firmware Emulator




remove files


Fix some parsing errors


Formatting


Change rpm_per_tick

* Tests

* Modify home functionality


Formatting and linting

* fix rpm emulator g-code and add deactivate test

* Fix tests

* Add deactivate_heater G-Code

* Add start_set_temperature

* Update home and deactivate_heater functionality

Based off of https://github.com/Opentrons/opentrons/pull/11121#issuecomment-1190383124

* Linting

* fix: Fix deactivate_heater function

* Make home_delay_time customizable

* Linting

Co-authored-by: jbleon95 <jeremybenleon@gmail.com>

Co-authored-by: jbleon95 <jeremybenleon@gmail.com>

andySigler committed:
fix(api): be prepared for aspirate after pick-up-tip (#10914)

* might as well prepare for aspirate right after picking up a tip...

* prepare-for-aspirate within HWApi's pick-up-tip methods; fixes tests

* adds prep_after=True to hardware API.pick_up_tip()

* adds prep_after kwarg to .pick_up_tip()

* pipette simulation uses prep_after arg

* adds more to prep_after test in hw api

* adds test for new pick_up_tip API arg prep_after

* bumbs API version to 2.13

* updates docstrings

* adds tests for comparing 2.12 vs 2.13 pick-up-tip behavior

* Update api/src/opentrons/protocol_api/instrument_context.py

Co-authored-by: Ed Cormany <ecormany@gmail.com>

* Update api/src/opentrons/protocol_api/instrument_context.py

Co-authored-by: Ed Cormany <ecormany@gmail.com>

* g-code-testing: update 2_modules

* g-code-testing: update omega

* g-code-testing: update swift turbo

* g-code-testing: update swift smoke

* update the rest of the gcodes

Co-authored-by: Ed Cormany <ecormany@gmail.com>
Co-authored-by: Seth Foster <seth@opentrons.com>

jerader committed:
refactor(app): refactor useRunStatuses and remove unnecessary logic (#11206)

closes #11199

koji committed:
refactor(app): add new spacings and replacing (#11202)

This PR will add 2 new spacing and replace old spacing with new spacing const
fix #11181

shlokamin committed:
feat(app): use started at timestamp in run log (#11213)

SyntaxColoring committed:
fix(api,update-server): Synchronize robot name more tightly between between api and update-server (#11175)

jerader committed:
refactor(app): h-S troubleshooting guide links to correct pages (#11196)

pmoegenburg committed:
feat(api): heater-shaker: error status setting (#10765)

* feat(api): heater-shaker: error status setting

When an error response comes from the heater-shaker (including in the
background polls, since the module firmware can initiate asynchronous
error states), in addition to passing on any exceptions we need to set a
longer-lasting error state that can be retrieved and reset by higher
levels.

Closes #9644

* added first test

* formatted

* fixed undefined 'opentrons'

* updated patch target

* imported mock for AsyncMock

* disabiguated imports

* fixed patch semantics

* formatted

* successfully implemented tests

* removed unused fixture

* added synchronous tests

* updated tests and error guarding function

* formatted

* formatted

* linting updates

* removed _guard on synchronous commands

* fixed linting errors

* refactored synchronous test

* moved error state to Listener and updated sync and async tests

* updated test with more appropriate error

* reverted function call order for cleaner diff

* Once more. Also updated tests with to raise Exception instead of RuntimeError

* proposed fix to catch error responses in _send_data

* fixed _send_data

* lint format

* updated read_until call

* updated read_until

* reverted HS driver ACK

Co-authored-by: Seth Foster <seth@opentrons.com>

pmoegenburg committed:
feat(heater-shaker): add firmware upload capability (#10307)

* added upload_via_dfu method and updated find_bootloader_port for stm32 modules

* formatted

* fix for testing

* fix for testing

* Made PR requested changes and added test

* formatted

* fixes from linting

* updated tests

* added test

* updated tests

* updated the find-and-enter-bootloader flow for h/s

* don't use returncode for long processes

* linted, updated tests, and made some suggested changes

* halp

* formatted

* updated locking mechanism

* removed comments

Co-authored-by: Sanniti <sanniti@opentrons.com>

mcous committed:
test(protocol-engine): broaden acceptance criteria of wait for duration tests (#11230)

y3rsh committed:
chore(issue templates): update and use form syntax for issue templates (#11222)

* chore(issue templates): update and use form syntax for issue templates

Remove the 6.0 feedback testing form and use the new form syntax for the existing bug and feature
request issue templates.

* tweak format and add reques for app version

* lint fix

* Update .github/ISSUE_TEMPLATE/feature_request.yml

Co-authored-by: Mike Cousins <mike@opentrons.com>

* Update .github/ISSUE_TEMPLATE/bug_report.yml

Co-authored-by: Mike Cousins <mike@opentrons.com>

jerader committed:
feat(app): update React Select to 5.4.0 and add to App package.json (#11104)

follow up PR to close #11020

koji committed:
refactor(app): replace Text with StyledText (#11205)

This PR will replace Text with StyledText and remove color={COLORS.darkBlack} which is the current
default text color. fix #11149

mcous committed:
fix(api): if needed, home both pipettes with InstrumentContext.home (#11072)

Fixes #7499

jerader committed:
refactor(app): h-S refactor useLatchControls hook send command logic (#11238)

shlokamin committed:
refactor(shared-data): update HS labware defs to match updated measurements (#11214)

* refactor(shared-data): update HS labware defs to match updated dimension measurements

shlokamin committed:
refactor(step-generation): fix HS pipetting restrictions (#11212)

* refactor(step-generation): fix HS pipetting restrictions

sanni-t committed:
feat(api): implement pipetting restrictions around heater-shaker (#11218)

Co-authored-by: Max Marrone <max@marrone.nyc>

smb2268 committed:
fix(components): fix well highlight so it matches prod on PD (#11240)

koji committed:
fix(app): this will close #11223 (#11231)

This PR will fix the overflow menu disable state and when a robot is reachable, make an overflow menu
disabled. fix #11223

SyntaxColoring committed:
fix(api,shared-data): Fix incorrect rejection of `thermocycler/runProfile` commands (#11241)

DerekMaggio committed:
feat(g-code-parsing):  Add ability to run multiple PAPI versions  (#11221)

* feat(g-code-parsing): Add ability to run multiple PAPI versions

* Add results for PAPI version 2.12 and 2.13

* Overhaul cli to allow for selection of individual PAPI versions

* Add functionality to check for missing configuration files

* Sort returned configurations

* Check for missing files before running

* Formatting

* Add documentation on adding new protocols and comparison files

* Hardcode results_dir


More

* Remove S3 language

* Add globbing to cli

* Add api version arg

* Add blurbs to all protocols about apiLevel being ignored

* Update README

* Format README

* Add derived naming

shlokamin committed:
refactor(app): disable sending shake command when input out of range (#11237)

shlokamin committed:
refactor(app): LPC stop shaking and close labware latch (#11156)

closes #11126

DerekMaggio committed:
chore: Fix deprecated license field (#11249)

Field was causing errors when trying to build projects with poetry

ahiuchingau committed:
refactor(OT3): make gripper z its own axis kind (#11217)

jerader committed:
feat(app): support module JSON v6 commands in run log (#11234)

closes #11228

MichaelMulieri committed:
refactor(app): remove brackets of string attributes (#11242)

Closes #11211.

vabruzzo committed:
feat(app): add customLabware to labware filterBy (#11236)

* feat(app): add customLabware to labware filterBy

Closes #10917.

* add test

Co-authored-by: Vincent Abruzzo <vincent.abruzzo@opentrons.com>

vabruzzo committed:
refactor(app): rm need to create new components on every render of app tab content (#11224)

smb2268 committed:
feat(app): adding support for v6 commands in run log (#11254)

fix #11247
Co-authored-by: Jethary Rader <jethary@opentrons.com>

jbleon95 committed:
feat(engine): move pipettes away if blocking heater-shaker open latch or start shake (#11248)

ecormany committed:
docs: fix PD support page link (#11258)

smb2268 committed:
feat(app): update modal component and labware liquid modal (#11216)

fix #11142

jerader committed:
refactor(app): clean up json v6 command text in run log components (#11259)

closes #11253

shlokamin committed:
refactor(app): close labware latch before setting shake speed (#11262)

shlokamin committed:
refactor(app): remove bad LPC i18n string call (#11239)

jerader committed:
test(app): create tests for ProtocolsLanding and ProtocolList (#11265)

closes #11219

sfoster1 committed:
fix(api): do not acquire gpio in ot3simulator (#11251)

Because it shouldn't try, since it's a simulator, and it needs to not crash

shlokamin committed:
refactor(app): put LPC error string in i18n (#11271)

shlokamin committed:
feat(step-generation, shared-data): delete TC await profile complete command (#11273)

closes #11272

brenthagen committed:
refactor(app): delete unused 5.0 app code (#11198)

shlokamin committed:
feat(robot-server): allow live commands when run is current and unstarted (#11267)

shlokamin committed:
refactor(step-generation): raise latch open error if latch state is null (#11274)

jerader committed:
refactor(app): HS refactor Wizard to only issue live commands (#11245)

closes #11250

MichaelMulieri committed:
refactor(app): replace unnecessary string interpolation (#11256)

shlokamin committed:
feat(protocol-designer): add slot 6 as accepted slot for HS (#11275)

closes #11148

smb2268 committed:
fix(app): add support for labware on modules in run log (#11276)

sanni-t committed:
fix(engine): thermocycler movement flagger in analysis (#11261)

Closes #11235

sfoster1 committed:
fix(hardware): handle timeout finding pipettes (#11252)


Without doing this, starting the robot server without pipettes attached
won't work.

This required a bit of a deeper refactor to return partially-successful results.

mcous committed:
fix(api): inlcude command error details in top-level fatal error (#11277)

Re: RSS-43

jerader committed:
refactor(app): update pipette mismatch text in setup to match design system (#11278)

closes #11266 and #RAUT-71

Commit:
refactor(app): update H-S running modal command to stop active H-S modules (#11244)

* refactor(app): update H-S running modal command to stop active H-S modules

This PR will use the HS running modal before starting a run to stop all active heater shakes. It also fixes a bug where the wrong heater shaker was rendered when opening the wizard from the setup page.

closes #11194

sanni-t committed:
feat(papiv2): retract pipettes before shaking or opening latch (#11268)

Closes #11210, closes #11209

Co-authored-by: Mike Cousins <mike@opentrons.com>

koji committed:
feat(app): replace lookup with search with google in electron-context-menu (#11232)

This PR will replace lookup with search google in electron-context-menu.
fix #11220 and RAUT-66

shlokamin committed:
feat(protocol-designer): render modals within viewport (#11284)

shlokamin committed:
refactor(protocol-designer): remove HS feature flag (#11282)

shlokamin committed:
refactor(protocol-designer): throw timeline error when heater shaker module id is missing (#11287)

closes #11280

jerader committed:
refactor(protocol-designer): add fixed position to modals (#11285)

Commit:
refactor(protocol-designer): move liquids UI enhancements behind feature flag (#11289)

This PR moves the liquids UI enhancements behind the feature flag.

TamarZanzouri committed:
refactor(api): remove heater shaker feature flag (#11283)

vabruzzo reviewed (approved):
🎉

koji committed:
fix(app): fix invalid dom nesting issue on labware (#11294)

This will fix the invalid dom nesting issue <p><div></div></p> and close RAUT-79

koji committed:
fix conflicts

koji committed:
fix conflicts

koji committed:
Merge branch 'chore_app-update-storybook' of https://github.com/Opentrons/opentrons into chore_app-update-storybook

koji commented:
Seems that I messed up this branch when I tried to resolve the conflicts.
I will open another PR shortly for upgrade Storybook.
